### ISSUE TITLE
Vector Array and Arduino Communication Protocol

### ISSUE THREAD

**Issue by mmansion** [mmansion/DrawingMachines#3]
@kuanju feel free to edit this as needed. 

*Processing*

* [ ] Function called `runSketch(PVector[] coords)` that takes an array of PVector() objects
* [ ] `runSketch()` function should iterate, and send one coordinate one at a time
* [ ] `runSketch()` loop should wait for each sent coord to be acknowledged and then run before sending the next

*Arduino*

* [ ] Firmware should handle incoming coordinates and move robot to that position
* [ ] Firmware should callback to Processing when move is complete, expecting another coordinate to be sent

**Comment by mmansion** [comment]
I think the simulator should eventually get built right into the controller app and provide 2 functions: "run" and "simulate". 

I'll work on the simulator and try to use the same function `runSketch()`, having it work in a similar way to the real thing. The only difference when simulating is that it won't actually go out through the serial port.

I have created a branch called [SKittBotController](https://github.com/mmansion/DrawingMachines/tree/SkittBotController) for this purpose.

I will continue to merge your updates from serial-communication into that branch.

**Comment by kuanju** [comment]
cool. Going to be offline until 830pm. Will pick up the left then. ;)

**Comment by mmansion** [comment]
I don't think we need to rush on this before tomorrow. We can still evolve it over the next couple weeks.

**Comment by mmansion** [comment]
FYI: I started working on the processing end of things in order to get the simulator to draw using SCODE coordinates (SKittBot Code)
